### DISCUSSION TITLE
Initiator fallback to a different IP family

### DISCUSSION THREAD

**Discussion by robdyck** [strongswan/strongswan#2122] [Ideas]
I encountered a road warroir ( RW ) situation where the server FQDN resolves to both an IPV4 and IPV6 address. Strongswan tries only the IPV6 address. The mobile provider provides only IPV4 continuity so the connection fails. This happens with the desktop version as well when tethered to the phone. I couldn't find a work around in the configuration other than IPV4 only.
The server has a dynamic address. In my situation I control the DNS and can create an IPV4 only FQDN. Not everyone can do this.

My idea is that strongswan implement a continuity check prior to attempting a connection. A less satisfactory solution would a fallback to a different IP family after a time period.

**Comment by tobiasbrunner** [comment]
> Strongswan tries only the IPV6 address.

It tries the first address returned by the resolver. It currently can't fallback to other addresses that might be returned.

> This happens with the desktop version as well when tethered to the phone.

Not sure what you mean by that.

> I couldn't find a work around in the configuration other than IPV4 only.

Not sure what you mean here either. Are you referring to disabling `charon.plugins.socket-default.use_ipv6`? Or configuring `%any4` or a specific IPv4 address as local address? Or to changing the DNS records?

What exactly does happens in your case? Can strongSwan resolve a local address? Are packets sent? Do you have a log?
